lib/instances/BaseInstance.lua

```diff
@@ -180,6 +180,27 @@ function BaseInstance.prototype:GetChildren()
 	return result
 end
 
+-- GetDescendants implementation using a queue rather than recursion
+function BaseInstance.prototype:GetDescendants()
+	local queue = {}
+	local descendants = {}
+	local current = self
+
+	while current do
+		local children = current:GetChildren()
+
+		for _,child in pairs(children) do
+			descendants[#descendants + 1] = child
+			queue[#queue + 1] = child
+		end
+
+		current = queue[#queue]
+		queue[#queue] = nil
```

Comment:
Does Roblox's `GetDescendants` have any ordering guarantees? In general, I don't mind if we break them, but it might be something to keep in mind.

Reply:
https://wiki.roblox.com/index.php?title=API:Class/Instance/GetDescendants

Apparently it does, "The arrays returned by GetDescendants are arranged so that parents come earlier than their children."

I can't imagine anyone relies on this behavior though.